api/v1beta1/openstackdataplanenodeset_types.go

```diff
@@ -238,3 +246,56 @@ func (r *OpenStackDataPlaneNodeSetSpec) duplicateNodeCheck(nodeSetList *OpenStac
 
 	return
 }
+
+// Compare TLS settings of control plane and data plane
+// if control plane name is specified attempt to retrieve it
+// otherwise get any control plane in the namespace
+func (r *OpenStackDataPlaneNodeSetSpec) ValidateTLS(namespace string) error {
+	var err error
+
+	controlPlane := openstackv1.OpenStackControlPlane{}
+	if len(r.ControlPlane) > 0 {
+		controlPlaneName := types.NamespacedName{
+			Name:      r.ControlPlane,
+			Namespace: namespace,
+		}
+		err = webhookClient.Get(context.TODO(), controlPlaneName, &controlPlane)
```

Comment:
You've changed the code to call this function from the controller - but you're still using webhook related variables here.  What happens if someone decides to disable webhooks ?  -- you'll end up with an error here.  Instead this should have the ctx and client passed in -- and maybe even should be defined in the controller file itself - like other such functions.